### PR TITLE
feat: Mobility operator filter

### PR DESCRIPTION
**Pull request by KristianSelnas**

### CODE REVIEW

File: src/api/mobility.ts
Comment (jorelosorio) [inline src/api/mobility.ts:34]: Is this something to cover on this PR? inject formFactor from outside?

File: src/api/mobility.ts
Comment (jorelosorio) [inline src/api/mobility.ts:86]: We could check if the `stations` if bigger than 0.

Comment (jorelosorio) [reply src/api/mobility.ts:86]: Or `res.data.stations?.[0]`

File: src/api/mobility.ts
Comment (jorelosorio) [inline src/api/mobility.ts:85]: `stringifyUrl(url, query), opts` might work as well. instead of `{...opts}`

File: src/api/mobility.ts
Comment (jorelosorio) [inline src/api/mobility.ts:71]: See the comment below ;)

File: src/mobility/use-operators.tsx
Comment (jorelosorio) [inline src/mobility/use-operators.tsx:11]: Mmm, I wonder, If it will be at the end an array why not just (types?: FormFactor[])

**Review by jorelosorio — CHANGES_REQUESTED** [review]

**Review by gorandalum — APPROVED** [review]

File: src/api/mobility.ts
Comment (KristianSelnas) [reply src/api/mobility.ts:34]: It's an old TODO. It may appear to be added now, since this code has been moved from two files to one. It will be resolved as soon as other vehicles than scooters are added to the app. (bikes and cars are _stations_)

File: src/api/mobility.ts
Comment (KristianSelnas) [reply src/api/mobility.ts:71]: Yes, this is strictly not necessary, but really convenient when you need to test against a local BFF an have to add `baseUrl`:
```
{
    ...opts,
    baseUrl: 'http://localhost:8080'
}
```
I'll remove it, nevertheless. All minor things like this will eventually add up

File: src/mobility/use-operators.tsx
Comment (KristianSelnas) [reply src/mobility/use-operators.tsx:11]: It's just more convenient from where this function is called. E.g. I can now do like this `useOperators(FormFactor.Scooter)` rather than `useOperators([FormFactor.Scooter])`.

File: src/mobility/components/filter/MobilityFilters.tsx
Comment (KristianSelnas) [reply src/mobility/components/filter/MobilityFilters.tsx:17]: Hmm... I see your point. Good spot 👍  React noob mistake.

Not sure if I see the best solution, though. Ideally, I don't want all details about mobility filter stuff to bleed all the way out to `MapFilters`.

Comment (KristianSelnas) [reply src/mobility/components/filter/MobilityFilters.tsx:17]: Any suggestions?

Comment (KristianSelnas) [reply src/mobility/components/filter/MobilityFilters.tsx]: I've made a change suggestion now (see my latest commit, https://github.com/AtB-AS/mittatb-app/pull/3596/commits/ef0b6069fc882fee3041abe55746af3e0c2c329b).

I don't think it's too bad. Updating the `MapFilterType` should probably be owned by `MapFilterSheet` anyways.

Comment (gorandalum) [reply src/mobility/components/filter/MobilityFilters.tsx]: > I've made a change suggestion now

I think it looks good. A little more verbose, but it is more correctly implemented. 

Another strategy for cases like this is to split it into a React Component and hook. That would mean `MobilityFilters` only has the responsibility to show the toggles, and sync the toggle state to the parent. And then it would be a `useMobilityFilters` (or something like it) which takes in the toggle states and has the responsibility for applying them correctly and returning the state with the correct vehicles. Not necessarily better to do it like this here now, but just mentioning it.

> Any suggestions?

Just reach out on Slack if something should be discussed right away. I try to take a look at PRs at morning and after lunch, so will not always respond to comments immediately.

Comment (KristianSelnas) [reply src/mobility/components/filter/MobilityFilters.tsx]: Ok, I'll leave it like this for now. 

I'll wait until after the release branch has been made to merge this. It requires both BFF and config changes, and it is strictly a part of next sprint anyways.

**Review by jorelosorio — APPROVED** [review]

**Pull request merged**
